docs/usage/upload_files.rst

```diff
@@ -0,0 +1,199 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+
+Creating Buckets
+----------------
+
+In order to upload, modify or delete files, we first need to create a bucket.
+A bucket can be created by a POST request to the url prefix, i.e. /files.
+The response will contain the unique ID of the bucket.
+A bucket can have one or more tags which store extra metadata for that bucket.
```

Comment:
Maybe explain what is a tag: extra metadata to "attach" extra information and describe a bucket. It is a loose key/value.